update docs on headings (#305)
wpaccessibility/wp-a11y-docs: docs/topics/content/headings/index.md

---
title: Headings
layout: default
parent: Content and images
description: A heading should simply and concisely describe the content that follows.
nav_order: 3
---

# Using headings in the content

## What are headings?

Headings are the framework of your content. A good heading should simply and concisely describe the content that follows.

> The heading level is important. The H1 (heading level 1) is reserved for the post or page title. For most themes that is the title you give a page or post. 

Reviewer: ```suggestion
The heading level is important. The H1 (heading level 1) is reserved for the post or page title. In most themes that is the title you give a page or post. 
```

![Screenshot adding a page title in the WordPress Admin]({{site.baseurl}}/assets/images/add-title.png)

> Use other heading levels H2 upto H6 to divide your content into meaningful sections, like the index of a book.

Reviewer: ```suggestion
Use other heading levels from `h2` to `h6` to divide your content into meaningful sections, like the index of a book.
```

> Choose heading levels by their meaning, not by their visible size. This way artificial intelligents (AI), search engines and screen readers understand the structure and meaning of the content better. 

Reviewer 2: ```suggestion
Choose heading levels by their meaning, not by their visible size. This way, artificial intelligence (AI) agents, search engines and screen readers understand the structure and meaning of the content better. 
```

{: .info .callout }
How to implement headings in development is addressed in the “Frontend code” section [Heading structure in theme development]({{site.baseurl}}/docs/topics/code/heading-structure/).

### Best practices for headings summarized

- The H1 (heading level 1) is reserved for the post or page title.
- Use the other headings to divide content into scannable blocks.
- Use headings to describe the content below. Do not use a heading just to make the text appear bigger or stand out.
- Don't skip a heading level from the top down.

> - Use a real HTML heading instead of making text only bold or large.

Reviewer: ```suggestion
- Use real HTML heading elements instead of making text bold or large.
```

Reviewer: This last point is a little tricky. You actually *should* just make text bold or large if it doesn't function as a heading. This is advice that leads to people using headings every time they want large text. Maybe:

"Use real HTML heading elements instead of making text bold or large, unless the text does not describe the content that follows it."

> A heading means “this text describes the information that follows it.” There should not be text before the heading that is part of what that heading describes, and there must be content following the heading.

Reviewer: Is this really part of the summary? I was sort of expecting the summary to just be the bullet points. Should this be moved above the summary heading?

## Examples of heading structures

Your page’s headings might resemble a structure similar to this one on a complex page:

```text
Heading 1 (post title)
    Heading 2 (major section)
        Heading 3 (subsection)
    Heading 2 (additional major section)
        Heading 3
            Heading 4
        Heading 3
            Heading 4
    Heading 2
        Heading 3 (post sub-heading)
        Heading 3 (post sub-heading)
 ```

If we use this post as an example, the heading structure for the content looks like this:

```text
    Heading 1: Using Headings in the content (main post topic)
        Heading 2: What are headings? (subsection)
            Heading 3: Best practices for headings summarized
        Heading 2: Examples of heading structures (subsection)
        Heading 2: How to test for a proper heading structure (subsection)
        Heading 2: Resources (subsection)
            Heading 3: WCAG Success Criteria for the heading structure
            heading 3: Other resources
```

## How to test for a proper heading structure

> In WordPress, you can check the heading structure in the  "Outline" tab. The outline tab is visible after clicking the button "Document Overview" in the top bar of the editor.

Reviewer: I'm going to open an issue for this; but I wonder if we need to have some styles that allow us to display the visual appearance of the icons used for some of these buttons, to make these instructions easier to understand.

Author: 👍

![Screenshot of the document outline of the page get started at a contributor day]({{site.baseurl}}/assets/images/document-outline.png)

Use the browser tool **HeadingsMap** to test the heading structure of an entire web page:
- [Headings Map toolbar for Edge](https://microsoftedge.microsoft.com/addons/detail/headingsmap/bokekiiaddinealohkmhjcgfanndmcgo)
- [Headings Map toolbar for Chrome](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi?hl=en)
- [Headings Map toolbar for Firefox](https://addons.mozilla.org/en-US/firefox/addon/headingsmap/)

{: .callout .info }

> **Note**: If you use a headings validation tool, a warning may pop up when the first heading is not an H1. This only applies when your web page needs to comply with WCAG 2 AAA (triple A). The global web standard is WCAG 2 AA (double A) and starting with an H2 (for e.g. the navigation) is allowed at the AA standard.

Reviewer: ```suggestion
**Note**: If you use a headings validation tool, a warning may pop up when the first heading is not an H1. This only applies when your web page needs to comply with WCAG 2 AAA (triple A). The global web standard is WCAG 2 AA (double A) and starting with an H2 (such as in the navigation) is allowed at the AA standard.
```
